### PR TITLE
fix commons fileupload usage and remove httpclient

### PR DESCRIPTION
**Pull request by msladek**
https://synjira.atlassian.net/browse/CELDEV-812

### CODE REVIEW

File: celements-xwiki-core/src/main/java/com/xpn/xwiki/XWiki.java [hunk 1]
```diff
@@ -588,49 +581,6 @@ public static String getFormEncoded(String content) {
     return filter.process(content);
   }
 
-  public static HttpClient getHttpClient(int timeout, String userAgent) {
```
Comment (fpichler) [inline celements-xwiki-core/src/main/java/com/xpn/xwiki/XWiki.java:591]: 👍  https://i.gifer.com/9MNb.gif

File: celements-xwiki-core/src/main/java/com/xpn/xwiki/XWiki.java [hunk 2]
```diff
@@ -5564,128 +5514,6 @@ public String getHttpUserAgent(XWikiContext context) {
     }
   }
 
-  public String getURLContent(String surl, XWikiContext context) throws IOException {
```
Comment (fpichler) [inline celements-xwiki-core/src/main/java/com/xpn/xwiki/XWiki.java:5567]: 👍 https://i.gifer.com/9MNb.gif

File: celements-xwiki-core/src/main/java/com/xpn/xwiki/api/XWiki.java [hunk 3]
```diff
@@ -2503,123 +2503,6 @@ public String getAdClientId() {
     return this.xwiki.getAdClientId(getXWikiContext());
   }
 
-  /**
```
Comment (fpichler) [inline celements-xwiki-core/src/main/java/com/xpn/xwiki/api/XWiki.java:2506]: 👍 https://i.gifer.com/9MNb.gif

**Review by fpichler — APPROVED** [review]

**Comment by msladek** [comment]
@fpichler chop chop :)

**Pull request merged**